metrics/internal/collectors/ceph-blocklist.go

```diff
@@ -45,21 +65,32 @@ func (c *CephBlocklistCollector) Collect(ch chan<- prometheus.Metric) {
 	c.PersistentVolumeStore.Mutex.RLock()
 	defer c.PersistentVolumeStore.Mutex.RUnlock()
 
-	for node, clients := range c.PersistentVolumeStore.RBDClientMap {
-		for _, watcher := range clients.Watchers {
-			ip, port, nonce := parseAddress(watcher.Address)
-			if ip == "" {
-				klog.Errorf("error parsing address %s", watcher.Address)
-				continue
-			}
+	blockedNodes, err := c.getNodes()
+	if err != nil {
+		klog.Errorf("failed to get nodes: %v", err)
+	}
 
+	for client, nodes := range c.PersistentVolumeStore.RBDClientMap {
+		ip, port, nonce := parseAddress(client)
+		if ip == "" {
+			klog.Errorf("error parsing address %s", client)
+			continue
+		}
+		for _, node := range nodes {
 			if c.CephBlocklistStore.IsBlocked(ip, port, nonce) {
-				ch <- prometheus.MustNewConstMetric(c.NodeRBDBlocklist, prometheus.GaugeValue, 1, node)
-			} else {
-				ch <- prometheus.MustNewConstMetric(c.NodeRBDBlocklist, prometheus.GaugeValue, 0, node)
+				blockedNodes[node] = true
+
 			}
 		}
 	}
+
+	for node, blocked := range blockedNodes {
+		if blocked {
+			ch <- prometheus.MustNewConstMetric(c.NodeRBDBlocklist, prometheus.GaugeValue, 1, node)
+		} else {
+			ch <- prometheus.MustNewConstMetric(c.NodeRBDBlocklist, prometheus.GaugeValue, 0, node)
+		}
+	}
```

Comment:
Do we need to run this loop? Would it be possible to create these metrics in the above loop when you're updating the `blockedNodes` map?